otelcol.auth.headers: new component (#2436)
grafana/agent: docs/sources/flow/reference/components/otelcol.auth.headers.md

---
aliases:
- /docs/agent/latest/flow/reference/components/otelcol.auth.headers
title: otelcol.auth.headers
---

# otelcol.auth.headers

`otelcol.auth.headers` exposes a `handler` that can be used by other `otelcol`
components to authenticate requests using custom headers.

> **NOTE**: `otelcol.auth.headers` is a wrapper over the upstream OpenTelemetry
> Collector `headerssetter` extension. Bug reports or feature requests will be
> redirected to the upstream repository, if necessary.

Multiple `otelcol.auth.headers` components can be specified by giving them
different labels.

## Usage

```river
otelcol.auth.headers "LABEL" {
  header {
    key   = "HEADER_NAME"
    value = "HEADER_VALUE"
  }

>   token = "TOKEN"

Reviewer: ```suggestion

```

Author: doh, thanks :)

---
aliases:
- /docs/agent/latest/flow/reference/components/otelcol.auth.headers
title: otelcol.auth.headers
---

# otelcol.auth.headers

`otelcol.auth.headers` exposes a `handler` that can be used by other `otelcol`
components to authenticate requests using custom headers.

> **NOTE**: `otelcol.auth.headers` is a wrapper over the upstream OpenTelemetry
> Collector `headerssetter` extension. Bug reports or feature requests will be
> redirected to the upstream repository, if necessary.

Multiple `otelcol.auth.headers` components can be specified by giving them
different labels.

## Usage

```river
otelcol.auth.headers "LABEL" {
  header {
    key   = "HEADER_NAME"
    value = "HEADER_VALUE"
  }
  token = "TOKEN"
}
```

## Arguments

> `otelcol.auth.headers` doesn't support any arugments and is configured fully
> through inner blocks.

Reviewer 2: ```suggestion
`otelcol.auth.headers` doesn't support any arguments and is configured fully
through inner blocks.
```

`value` | `string` or `secret` | Value of the header. | | no
`from_context` | `string` | Metadata name to get header value from. | | no

Exactly one of `value` or `from_context` must be provided for each `header`
block.

The `value` attribute sets the value of the header directly.

Alternatively, `from_context` can be used to dynamically retrieve the header
value from request metadata.

> **NOTE**: It is not possible to use `from_context` to get the header value if
> [the `otelcol.processor.batch` component][otelcol.processor.batch] is used to
> batch before data is sent to the component referencing
> `otelcol.auth.headers`.

[otelcol.processor.batch]: {{< relref "./otelcol.processor.batch.md" >}}

## Exported fields

The following fields are exported and can be referenced by other components:

Name | Type | Description
---- | ---- | -----------
`handler` | `capsule(otelcol.Handler)` | A value that other components can use to authenticate requests.

## Component health

`otelcol.auth.headers` is only reported as unhealthy if given an invalid
configuration.

## Debug information

`otelcol.auth.headers` does not expose any component-specific debug information.

## Example

> This example configures [otelcol.exporter.otlp][] to use headers token
> authentication:

Author: ```suggestion
This example configures [otelcol.exporter.otlp][] to use custom headers:
```
